src/contracts/confidentialVoting.ts

```diff
@@ -0,0 +1,137 @@
+import {
+    ByteString,
+    FixedArray,
+    HashedMap,
+    PubKey,
+    Sha256,
+    Sig,
+    SmartContract,
+    assert,
+    fill,
+    hash256,
+    method,
+    prop,
+    toByteString,
+} from 'scrypt-ts'
+
+export type Candidate = {
+    name: ByteString
+    votesReceived: bigint
+}
+
+export class ConfidentialVoting extends SmartContract {
+    @prop()
+    voters: FixedArray<PubKey, 3>
+
+    @prop(true)
+    voteCommits: HashedMap<PubKey, Sha256>
+
+    @prop(true)
+    voteRevealed: boolean
+
+    @prop(true)
+    candidates: FixedArray<Candidate, 2>
+
+    @prop()
+    VoteDeadline: bigint
+
+    @prop()
+    revealDeadline: bigint
+
+    @prop(true)
+    voteFinished: boolean
+
+    constructor(
+        voters: FixedArray<PubKey, 3>,
+        candidates: FixedArray<Candidate, 2>,
+        voteCommits: HashedMap<PubKey, Sha256>,
+        voteDeadline: bigint,
+        revealDeadline: bigint
+    ) {
+        super(...arguments)
+        this.voters = voters
+        this.voteRevealed = false
+        this.voteCommits = voteCommits
+        this.VoteDeadline = voteDeadline
+        this.revealDeadline = revealDeadline
+        this.voteFinished = false
+        this.candidates = fill(
+            {
+                name: toByteString(''),
+                votesReceived: 0n,
+            },
+            2
+        )
+    }
+
+    @method()
+    public vote(voter: PubKey, voteCommitment: Sha256, sig: Sig) {
```

Comment:
You don't check if the passed public key is in the set of chosen voters. Make the `voters` property a HashedSet and check if it's present.